pkg/apis/core/types.go

```diff
@@ -3674,6 +3693,17 @@ type ServiceSpec struct {
 	// +optional
 	ExternalTrafficPolicy ServiceExternalTrafficPolicyType
 
+	// InternalTrafficPolicy specifies if the cluster internal traffic
+	// should be routed to cluster-wide endpoints, node-local endpoints only
+	// or both with node-local ones preferred if there is any. "Cluster"
+	// routes internal traffic to a Service to all cluster-wide endpoints.
+	// "PreferLocal" will route internal traffic to node-local endpoints if
+	// one exists, otherwise it will fallback to the same behavior as
+	// "Cluster". "Local" routes traffic to node-local endpoints only,
+	// traffic is dropped if no node-local endpoints are ready.
+	// +optional
+	InternalTrafficPolicy ServiceInternalTrafficPolicyType
```

Comment:
This actually needs to be a string pointer, so we can drop the field when disabled.

Reply:
would being inconsistent with `externalTrafficPolicy` a concern?

Reply:
I don't think so, `externalTrafficPolicy` pre-dates the requirement that new string fields have to be pointers (so they can be nil-able and entirely removed from API responses).

Reply:
(from a user dealing with json/yaml, there should be no functional difference)